### PR TITLE
Add transaction classes

### PR DESCRIPTION
**Pull request by karoliineh**
Added most of the transaction classes, but still needs some work:

- [x] Implement `TransactionController`
- [x] `Transaction` class is missing the `sessionUser` field from the class diagram
- [x] Add transaction to `DemoData`
- [x] Create tests for `TransactionController`
- [x] Create the classes related to `TransactionLog`

Differences from the class diagram:
* The TransactionRecord class has Account field, but the Account has list of Transactions instead of list of TransactionRecords (Transactions does not have Account field). In code, Accound has list of TransactionRecords.

### CODE REVIEW

**Comment by karoliineh** [comment]
Done (referencing [this class diagram](https://raw.githubusercontent.com/opendemobank/diagrams/7a55fa715405f4dc22985b599139b34485be3b83/class/ClassDomain.svg?token=AKTBDV4GKGTXXTYL6P7ZTRTBV55YG)):

* Transaction
* TransactionStatus 
* Currency
* Direction
* From TransactionManager (TransactionController + tests)
  * getAll()
  * getById
  * getByUserId (different from getByAccount, because transaction has only SessionUser)
  * createTransaction
  
 TODO:
* In TransactionManager (TransactionController + tests) @toomastamm 
  * editTransaction
  * stronoTransaction
* Should we implement the getByAccountId() for the TransactionRecords somewhere, to be able to see a list of transactionrecords related to an account? There is no TransactionRecordManager class though. Should we add one or just write this method in TransactionManager instead? Or what should we do with that? @toomastamm @PauliusKu

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,8 @@
     <name>backend</name>
     <description>backend</description>
     <properties>
-        <java.version>17</java.version>
+        <java.version>16</java.version>
```
Comment (toomastamm) [inline pom.xml]: Why downgrade?

Comment (toomastamm) [reply pom.xml]: nvm fixed in later commit

File: src/main/java/com/opendemobank/backend/controller/TransactionController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class TransactionController {
+
+    @Autowired
+    TransactionsRepo transactionsRepo;
+
+    @Autowired
+    UsersRepo usersRepo;
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Transaction> getById(@AuthenticationPrincipal User currentUser, @PathVariable("id") long id) {
```
Comment (toomastamm) [inline src/main/java/com/opendemobank/backend/controller/TransactionController.java]: Add 
`@Parameter(hidden = true) `
to currentUser so api documentation doesnt show it.

eg. 
`@Parameter(hidden = true) @AuthenticationPrincipal final User currentUser`

File: src/main/java/com/opendemobank/backend/controller/TransactionController.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return new ResponseEntity<>(transactionsRepo.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("/user/{id}")
+    public ResponseEntity<List<Transaction>> getAllBySessionUserId(@AuthenticationPrincipal User currentUser, @PathVariable("id") long id) {
+        User user = usersRepo.findById(id);
+        if (user == null)
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+
+        if (currentUser.getId() != id)
```
Comment (toomastamm) [inline src/main/java/com/opendemobank/backend/controller/TransactionController.java]: admins shouldnt get unauthorized, check role

File: src/main/java/com/opendemobank/backend/controller/TransactionController.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        List<Transaction> transactions = transactionsRepo.findAllBySessionUserId(id);
+
+        return new ResponseEntity<>(transactions, HttpStatus.OK);
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Transaction> createTransaction(@AuthenticationPrincipal User currentUser, @RequestBody Transaction newTransaction) {
+        // if the session user is not the same as the one that is stated to have created the transaction
+        if (!Objects.equals(currentUser.getId(), newTransaction.getSessionUser().getId()))
```
Comment (toomastamm) [inline src/main/java/com/opendemobank/backend/controller/TransactionController.java:68]: might need to check admin here too as well, dont remember

**Review by toomastamm — APPROVED** [review]

**Pull request merged**